runtime/src/main/cpp/Memory.cpp

```diff
@@ -2598,6 +2631,10 @@ void freezeSubgraph(ObjHeader* root) {
   ContainerHeader* rootContainer = root->container();
   if (isPermanentOrFrozen(rootContainer)) return;
 
+  MEMORY_LOG("Run freeze hooks on subgraph of %p\n", root);
+
```

Comment:
Note that running freeze hooks here is not entirely correct too:
freezing can fail (see below). In this case WBRs would be transitioned into "pre-frozen" state and make cycles with non-frozen WBRs leak.
But I suggest leaving the implementation as is for now, because failing freeze is typically used during development (to debug unexpected freezing).
Let's just document this for now.

Reply:
By "document" I mean "leave a comment to the source code" of course, not the documentation for WBR :)